programs/bpf_loader/src/lib.rs

```diff
@@ -1227,125 +1223,136 @@ mod tests {
         program_account.borrow_mut().data_as_mut_slice()[0] = 0; // bad elf
         assert_eq!(
             Err(InstructionError::InvalidAccountData),
-            process_instruction(&loader_id, &instruction_data, &keyed_accounts),
+            process_instruction(&loader_id, &[], &instruction_data, &keyed_accounts),
         );
     }
 
     #[test]
     fn test_bpf_loader_invoke_main() {
         let loader_id = bpf_loader::id();
-        let program_key = solana_sdk::pubkey::new_rand();
+        let program_id = Pubkey::new_unique();
         let program_account =
             load_program_account_from_elf(&loader_id, "test_elfs/noop_aligned.so");
 
         // Case: No program account
-        let mut keyed_accounts: Vec<KeyedAccountTuple> = vec![];
+        let mut keyed_accounts = vec![];
         assert_eq!(
             Err(InstructionError::NotEnoughAccountKeys),
-            process_instruction(&loader_id, &[], &keyed_accounts),
+            process_instruction(&loader_id, &[], &[], &keyed_accounts),
         );
 
         // Case: Only a program account
-        keyed_accounts.push((false, false, &program_key, &program_account));
+        keyed_accounts.push((false, false, program_id, program_account));
         assert_eq!(
             Ok(()),
-            process_instruction(&program_key, &[], &keyed_accounts),
+            process_instruction(&loader_id, &[0], &[], &keyed_accounts),
         );
 
         // Case: Account not a program
         keyed_accounts[0].3.borrow_mut().set_executable(false);
         assert_eq!(
-            Err(InstructionError::InvalidInstructionData),
-            process_instruction(&loader_id, &[], &keyed_accounts),
+            Err(InstructionError::IncorrectProgramId),
```

Comment:
Notice: The error code changed as it was previously testing:
`process_instruction(&loader_id, &[], &[], &keyed_accounts)`
which is incorrect and was changed to:
`process_instruction(&loader_id, &[0], &[], &keyed_accounts)`

This happened because it originally tested with `&loader_id` instead of `&program_key` which the other tests used.